lib/ansible/executor/task_executor.py

```diff
@@ -497,7 +499,7 @@ def _squash_items(self, items, loop_var, variables):
                 self._task.args['name'] = name
         return items
 
-    def _execute(self, variables=None):
+    def _execute(self, variables=None, notify_scope='task'):
```

Comment:
I would turn `notify_scope` parameter into an instance variable, `self._notify_scope`. That way you don't have to pass it around.